nixos/modules/services/web-apps/grist-core.nix

```diff
@@ -0,0 +1,250 @@
+
+            GVISOR_FLAGS = mkOption {
+              type = str;
+              default = "-rootless -debug";
+              description = ''
+                The flags that are passed on to gVisor when creating a sandbox.
+              '';
+            };
+
+            GVISOR_AVAILABLE = mkOption {
+              type = str;
+              default = "1";
+              readOnly = true;
+              description = ''
+                Whether gvisor is available for Grist.
+              '';
+            };
+
+            TYPEORM_DATABASE = mkOption {
+              type = str;
+              default = "/var/lib/grist-core/db.sqlite";
+              description = ''
+                Database filename for sqlite or database name for other db types.
+              '';
+            };
+
+            TYPEORM_TYPE = mkOption {
+              type = enum [
+                "sqlite"
+                "postgres"
+              ];
+              default = "sqlite";
+              description = ''
+                Which database type to use for storage.
+              '';
+            };
+          };
+        };
+      default = { };
+      example = {
+        GRIST_DEFAULT_EMAIL = "example@example.com";
+      };
+      description = ''
+        Environment variables used for Grist.
+        See [](https://github.com/gristlabs/grist-core/tree/v1.3.2?tab=readme-ov-file#environment-variables)
+        for available environment variables.
+      '';
+    };
+
+    environmentFiles = mkOption {
+      type = with types; listOf path;
+      default = [ ];
+      description = ''
+        Environment files for secrets.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.grist-core = {
+      description = "Grist Core";
+
+      after = [
+        "network.target"
+      ] ++ lib.optional (cfg.settings.TYPEORM_TYPE == "postgres") "postgresql.service";
+
+      wants = [ "network.target" ];
+
+      path = [
+        pkgs.gvisor
+        cfg.pythonEnv
+      ];
+
+      environment = cfg.settings;
+
+      serviceConfig =
+        {
+          ExecStart = lib.getExe cfg.package;
+
+          DynamicUser = true;
+          User = cfg.user;
+          Group = cfg.group;
+
+          Restart = "always";
+
+          RuntimeDirectory = "grist-core";
+          StateDirectory = "grist-core";
+
+          ProtectHome = true;
+          ProtectSystem = "strict";
+          PrivateTmp = true;
+          PrivateDevices = true;
+          ProtectHostname = true;
+          ProtectClock = true;
+          ProtectKernelTunables = true;
+          ProtectKernelModules = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          NoNewPrivileges = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          RemoveIPC = true;
+          PrivateMounts = true;
+
+          EnvironmentFile = cfg.environmentFiles;
+        }
+        // (lib.optionalAttrs cfg.enableSandboxing {
```

Comment:
It is surprising how this option mixes gvisor and systemd sandboxing. Why is systemd sandboxing made conditional anyway?

Reply:
It was an attempt to circumvent the issue I've described here: https://github.com/NixOS/nixpkgs/pull/376176#issuecomment-2830312715

By sandboxing I could overwrite `/usr/bin/python3` with the python environment from the nix store